workflow/Workflow/workflowImplementation.go

```diff
@@ -0,0 +1,158 @@
+package workflows
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	model "github.com/E-Furqan/Food-Delivery-System/Models"
+	utils "github.com/E-Furqan/Food-Delivery-System/Utils"
+	"go.temporal.io/sdk/workflow"
+)
+
+// func (wFlow *Workflow) RegisterWorkflow(ctx workflow.Context, registrationData model.User) error {
+// 	option := workflow.ActivityOptions{
+// 		StartToCloseTimeout: time.Second * 5,
+// 		RetryPolicy: &temporal.RetryPolicy{
+// 			InitialInterval:    time.Second * 10,
+// 			MaximumInterval:    time.Second * 30,
+// 			MaximumAttempts:    3,
+// 			BackoffCoefficient: 2.0,
+// 		},
+// 	}
+// 	ctx := workflow.WithActivityOptions(ctx, option)
+
+// 	err := workflow.ExecuteActivity(ctx, wFlow.Act.RegisterCheckRole, registrationData).Get(ctx, &registrationData)
+// 	if err != nil {
+// 		return err
+// 	}
+// 	log.Print("workflow implementation activity:", registrationData)
+// 	err = workflow.ExecuteActivity(ctx, wFlow.Act.CreateUser, registrationData).Get(ctx, &registrationData)
+// 	if err != nil {
+// 		return err
+// 	}
+// 	log.Print("error user", err)
+// 	return nil
+// }
+
+func (wFlow *Workflow) ViewDriverOrdersWorkflow(ctx workflow.Context, driverID uint, token string) error {
+	option := utils.ActivityOptions()
+	ctx = workflow.WithActivityOptions(ctx, option)
+
+	// var user model.User
+	// err := workflow.ExecuteActivity(ctx, wFlow.Act.GetUser, userID).Get(ctx, &user)
+	// if err != nil {
+	// 	return err
+	// }
+	var UserOrders []model.UpdateOrder
+	log.Print("workflow implementation activity:", driverID)
+	err := workflow.ExecuteActivity(ctx, wFlow.Act.ViewOrders, driverID, token).Get(ctx, &UserOrders)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (wFlow *Workflow) OrderPlacedWorkflow(ctx workflow.Context, order model.CombineOrderItem, token string) error {
+	option := utils.ActivityOptions()
+	ctx = workflow.WithActivityOptions(ctx, option)
+	var message string
+
+	var items []model.Items
+	err := workflow.ExecuteActivity(ctx, wFlow.Act.GetItems, order, token).Get(ctx, &items)
+	if err != nil {
+		wFlow.SendEmail(ctx, order.UpdateOrder, utils.Cancelled, &message, token)
+		utils.Sleep(ctx)
```

Comment:
why are we using sleep?

Reply:
i noticed that if we don't use sleep we were not receiving the email and the workflow was ending so i added a small sleep window so that email can be sent